#!/usr/bin/env python
PACKAGE = "self_aiming"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("y_kp", double_t, 0, "Kp for y axis", 0.0, -15, 15)
gen.add("y_kd", double_t, 0, "Kd for y axis", 0.0, -15, 15)

gen.add("z_kp", double_t, 0, "Kp for z axis", 0.0, -15, 15)
gen.add("z_kd", double_t, 0, "Kd for z axis", 0.0, -15, 15)

gen.add("center_x", double_t, 0, "Image center x axis", 0.0, -2, 2)
gen.add("center_y", double_t, 0, "Image center y axis", 0.0, -2, 2)

# gen.add("kalman_vx_cov", double_t, 0, "kalman vx covariace", 0.1, 0, 10)
# gen.add("kalman_vy_cov", double_t, 0, "kalman vy covariace", 0.1, 0, 10)

# gen.add("vy_kp", double_t, 0, "Kp for y axis velocity", 0.0, -5, 5)
# gen.add("vz_kp", double_t, 0, "Kp for z axis velocity", 0.0, -5, 5)

exit(gen.generate(PACKAGE, "rm_tracking_pid", "tuning"))
